src/components/useContainer.ts

```diff
@@ -168,7 +170,7 @@ export function useContainer() {
 			openID4VCIClientsJson[config.credentialIssuerIdentifier] = openID4VCIClient;
 		}
 		return { openID4VCIClientsJson, openID4VPRelyingParty, httpProxy, openID4VCIHelper, credentialParserRegistry };
-	}
+	}, [api, keystore, trustedCredentialIssuers]);
```

Comment:
I'm not sure this will work correctly with how `api` and `keystore` work, since those use hook-based state management, which means every state change results in a new instance. The functions registered here would be holding on to stale references to old instances with outdated state.

I guess everything gets reconstructed from scratch with fresh state on page refresh, though, so perhaps it actually works out for these use cases that have a couple page navigations in the middle. But it's something we need to at least be aware of if it becomes relevant.

Maybe we should refactor `api` and `keystore` to work more like these "true" singleton services, but it'll make it more cumbersome to have view components react to their state changes since they'll be living outside the React component lifecycle.

Reply:
i am sure too, which the best approach. Feel free to use the current pr or the #348, everything is better!